### PR TITLE
MODINVSTOR-1045 - Supporting source in /inventory-hierarchy/updated-instance-ids

### PR DESCRIPTION
**Pull request by khandramai**
[MODINVSTOR-1045](https://issues.folio.org/browse/MODINVSTOR-1045) - Supporting source in /inventory-hierarchy/updated-instance-ids

Extending API /inventory-hierarchy/updated-instance-ids to support filtering by instance source.

### CODE REVIEW

**Comment by khandramai** [comment]
Hi @marc-johnson , could you please assist with review? Thanks!

**Comment by marcjohnson-kint** [comment]
> Hi @marc-johnson , could you please assist with review? Thanks!

I'm happy to try to help. I'm not very familiar with the hierarchical view. Thus my feedback might be fairly limited.

File: src/main/java/org/folio/rest/impl/AbstractInstanceRecordsAPI.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 import java.time.OffsetDateTime;
 import java.util.Map;
+import java.util.Objects;
```
Comment (NikitaSedyx) [inline src/main/java/org/folio/rest/impl/AbstractInstanceRecordsAPI.java:9]: is it required?

File: src/main/java/org/folio/rest/impl/InventoryHierarchyAPI.java [hunk 2]
```diff
@@ -38,12 +41,15 @@ public class InventoryHierarchyAPI extends AbstractInstanceRecordsAPI implements
   @Validate
   @Override
   public void getInventoryHierarchyUpdatedInstanceIds(String startDate, String endDate, boolean deletedRecordSupport, boolean skipSuppressedFromDiscoveryRecords,
-      boolean onlyInstanceUpdateDate, String lang, RoutingContext routingContext, Map<String, String> okapiHeaders,
+      boolean onlyInstanceUpdateDate, String source, String lang, RoutingContext routingContext, Map<String, String> okapiHeaders,
       Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
     if(StringUtils.isEmpty(startDate) && StringUtils.isEmpty(endDate)) {
       String sql = SQL_INITIAL_LOAD;
+      if (Objects.nonNull(source)) {
+        sql+=String.format(SQL_INSTANCES_SOURCE_FILTER, source);
```
[the remark this answers is not in the document]
Comment (khandramai) [reply src/main/java/org/folio/rest/impl/InventoryHierarchyAPI.java:49]: @julianladisch , changed source to ENUM type instead of String. Is it enough?

**Review by obozhko-folio — APPROVED** [review]

File: ramls/inventory-hierarchy.raml [hunk 3]
```diff
@@ -56,6 +56,11 @@ resourceTypes:
           type: boolean
           required: false
           default: true
+        source:
+          description: Instance source
+          type: string
+          enum: [ FOLIO, MARC ]
+          required: false
```
[the remark this answers is not in the document]
Comment (khandramai) [reply ramls/inventory-hierarchy.raml:63]: @julianladisch , this case I need to handle possible SQL injections just verifying this string is FOLIO or MARC in code?

File: src/main/resources/templates/db_scripts/inventory-hierarchy/updateRecordsViewFunctionToSupportSource.sql
[the remark this answers is not in the document]
Comment (khandramai) [reply src/main/resources/templates/db_scripts/inventory-hierarchy/updateRecordsViewFunctionToSupportSource.sql:1]: @julianladisch , why do we need to remove code? I understand migration process like incremental updating of existing SQL scripts. From this point of view I added updated SQL snippet updateRecordsViewFunctionToSupportSource.sql starting from version 26.0.1.

**Review by siarhei-charniak — APPROVED** [review]

**Pull request merged**